uisdk/src/androidTest/java/com/karhoo/uisdk/trip/adyenTrip/AdyenTripTests.kt

```diff
@@ -0,0 +1,544 @@
+package com.karhoo.uisdk.trip.adyenTrip
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.ActivityTestRule
+import androidx.test.rule.GrantPermissionRule
+import com.github.tomakehurst.wiremock.common.ConsoleNotifier
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration
+import com.github.tomakehurst.wiremock.junit.WireMockRule
+import com.karhoo.sdk.api.model.TripInfo
+import com.karhoo.uisdk.R
+import com.karhoo.uisdk.booking.booking
+import com.karhoo.uisdk.common.Launch
+import com.karhoo.uisdk.common.serverRobot
+import com.karhoo.uisdk.common.testrunner.UiSDKTestConfig
+import com.karhoo.uisdk.screen.trip.TripActivity
+import com.karhoo.uisdk.trip.trip
+import com.karhoo.uisdk.util.TestData
+import com.karhoo.uisdk.util.TestData.Companion.ADYEN_PROVIDER
+import com.karhoo.uisdk.util.TestData.Companion.ADYEN_PUBLIC_KEY
+import com.karhoo.uisdk.util.TestData.Companion.DRIVER_TRACKING
+import com.karhoo.uisdk.util.TestData.Companion.GENERAL_ERROR
+import com.karhoo.uisdk.util.TestData.Companion.MEDIUM
+import com.karhoo.uisdk.util.TestData.Companion.QUOTE_LIST_ID_ASAP
+import com.karhoo.uisdk.util.TestData.Companion.REVERSE_GEO_SUCCESS
+import com.karhoo.uisdk.util.TestData.Companion.SHORT
+import com.karhoo.uisdk.util.TestData.Companion.TRIP
+import com.karhoo.uisdk.util.TestData.Companion.TRIP_ARRIVED
+import com.karhoo.uisdk.util.TestData.Companion.TRIP_CANCELLED_BY_FLEET
+import com.karhoo.uisdk.util.TestData.Companion.TRIP_DER
+import com.karhoo.uisdk.util.TestData.Companion.TRIP_DER_NO_DRIVER_DETAILS
+import com.karhoo.uisdk.util.TestData.Companion.TRIP_DER_NO_NUMBER_PLATE
+import com.karhoo.uisdk.util.TestData.Companion.TRIP_DER_NO_VEHICLE_NUMBER_PLATE_AND_DRIVER_DETAILS
+import com.karhoo.uisdk.util.TestData.Companion.TRIP_POB
+import com.karhoo.uisdk.util.TestData.Companion.TRIP_STATUS_ARRIVED
+import com.karhoo.uisdk.util.TestData.Companion.TRIP_STATUS_CANCELLED_BY_FLEET
+import com.karhoo.uisdk.util.TestData.Companion.TRIP_STATUS_CANCELLED_BY_USER
+import com.karhoo.uisdk.util.TestData.Companion.TRIP_STATUS_DER
+import com.karhoo.uisdk.util.TestData.Companion.TRIP_STATUS_POB
+import com.karhoo.uisdk.util.TestData.Companion.VEHICLES_ASAP
+import com.schibsted.spain.barista.rule.flaky.AllowFlaky
+import com.schibsted.spain.barista.rule.flaky.FlakyTestRule
+import org.junit.After
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.runner.RunWith
+import java.net.HttpURLConnection.HTTP_CREATED
+import java.net.HttpURLConnection.HTTP_INTERNAL_ERROR
+import java.net.HttpURLConnection.HTTP_OK
+
+@RunWith(AndroidJUnit4::class)
+class AdyenTripTests : Launch {
```

Comment:
Not sure we need separate Adyen/Braintree trip tests. Definitely for booking but the trip logic is independent of PSP. @mohadian your thoughts?

Reply:
I did wonder that at the time...